tracing-subscriber/src/layer.rs

```diff
@@ -494,6 +494,12 @@ where
     }
 
     fn try_close(&self, id: span::Id) -> bool {
+        let subscriber = &self.inner as &dyn Subscriber;
+        let _guard = match subscriber.downcast_ref::<Registry>() {
+            Some(registry) => Some(registry.ref_guard(id.clone())),
+            None => None,
+        };
```

Comment:
also, this will need to be flagged with `#[cfg(feature = "registry")]; i don't think this code compiles with that feature disabled. so, something like:
```rust
#[cfg(feature = "registry")]
let subscriber = &self.inner as &dyn Subscriber;
#[cfg(feature = "registry")]
let _guard = ...;
```

Reply:
Good point, fixed.